File: clients/py/src/features/resolve_ens_contenthash.py
from typing import Any
from polywrap_client import PolywrapClient
from polywrap_client_config_builder import PolywrapClientConfigBuilder
from polywrap_core import UriPackage
from polywrap_web3_config_bundle import get_web3_bundle

from validators import validate_uri



def run_test_case(input: Any) -> None:
    uri = validate_uri(input)

    print(f"URI Authority: {uri.authority}")

    config = (
        PolywrapClientConfigBuilder()
        .add(get_web3_bundle())
        .build()
    )

    client = PolywrapClient(config)

    print(f"Resolving: {uri}")

    result = client.try_resolve_uri(uri)

    match result:
        case UriPackage():
            print("Received: package")
            print("Success!")
        case _:
            raise ValueError("Expected UriPackage")

File: clients/js/src/features/resolve_ens_contenthash.ts
import { Input } from "../input";

import {
  PolywrapClientConfigBuilder,
  PolywrapClient
} from "@polywrap/client-js";

export async function runTestCase(input: unknown): Promise<void> {
  const uri = Input.expectUri(input);

  console.log(`URI Authority: ${uri.authority}`);

  const config = new PolywrapClientConfigBuilder()
    .addDefaults()
    .build();

  const client = new PolywrapClient(config);

  console.log(`Resolving: ${uri}`);

  const result = await client.tryResolveUri({ uri });

  if (result.ok) {
    console.log(`Received: ${result.value.type}`);
    console.log("Success!");
  }
}
